### DISCUSSION TITLE
`devIndicators` doesn't work for app router

### DISCUSSION THREAD

**Discussion by DoneDeal0** [vercel/next.js#63977] [App Router]
Hello,

The `devIndicators` doesn't work for the `app` router. 
<hr/>

**Bug**
Add the following code to the `next.config.js` file, as described in the [official documentation](https://nextjs.org/docs/app/api-reference/next-config-js/devIndicators).

```js
  devIndicators: {
    buildActivity: true,
    buildActivityPosition: "top-right",
  },
```
<hr/>

**Current behaviour**
In dev mode, when navigating to a new page (either via `next/navigation` router or `next/link`), no loader is displayed while the page is being compiled. 

**Expected behaviour**
In dev mode, when navigating to a new page, a loader should appear while the page is compiling. So we know the app isn't freezing without having to look at the terminal.
<hr/>

**Info**
```js
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.3.0: Wed Dec 20 21:31:00 PST 2023; root:xnu-10002.81.5~7/RELEASE_ARM64_T6020
Binaries:
  Node: 20.11.0
  npm: 10.2.4
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.1.4
  eslint-config-next: 13.4.19
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: N/A
```
<hr/>

How to fix this? Also, I have added a `loading.tsx` file for each page, but they just won't show up.

**Comment by coron3r** [comment]
I am also having this issue, if I have a project mixed with pages dir and app dir, the devIndicator works in the pages dir even when navigation to a app dir.

**Comment by Maki-Db** [comment]
It was reported as issue: https://github.com/vercel/next.js/issues/54100 .
The issue was rejected because it was already marked as to Do in the source code (from 2022): https://github.com/vercel/next.js/blob/2d9875983ca61a195a60048bf0493494002256af/packages/next/client/app-next-dev.js .
